packages/dappmanager/hostScripts/security_update.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+# Exit on error
+set -e
+
+# VARIABLES
+LOG_DIR="/usr/src/dappnode/logs"
+LOG_FILE="$LOG_DIR/security_update.log"
+DATE=$(date)
+FILE_SOURCES_LIST="/etc/apt/sources.list"
+FILE_SECURITY_SOURCES_LIST="/etc/apt/security.sources.list"
+DOCKER_PACKAGES=($(dpkg --get-selections | grep -o 'docker-ce\s\|containerd.io\s\|docker-ce-cli\s'))
+DOCKER_PACKAGES_LENGTH=${#DOCKER_PACKAGES[@]}
+DOCKER_INSTALLATION_REPOSITORIES=false
+
+############
+#0.LOG-FILE#
+############
+
+function create_log_file () {
+  mkdir -p $LOG_DIR
+  echo -e "\e[32mStarting security updates: $DATE\e[0m" >> $LOG_FILE
+}
+
+##############
+#1.PRE-UPDATE#
+##############
+
+function pre_update () {
+  echo -e "\e[32mPre update check\e[0m" >> $LOG_FILE 
+  # Prevent needrestart interactive mode and debian frontend during upgrade: https://manpages.debian.org/testing/needrestart/needrestart.1.en.html
+  export NEEDRESTART_SUSPEND=true # If set to a non-empty value the apt-get(8) hook will not run needrestart after installing or updating packages.
+  export DEBIAN_FRONTEND=noninteractive
+
+  # Check docker exists
+  if ! command -v docker &> /dev/null; then
+    echo "docker not installed" 2>&1 | tee -a $LOG_FILE 
+    exit 1
+  fi
+
+  # Check installation method
+  if [[ "$DOCKER_PACKAGES_LENGTH" -eq 3 ]]; then
+    echo -e "docker installed through repositories" >> $LOG_FILE 
+    DOCKER_INSTALLATION_REPOSITORIES=true
+  else
+    echo -e "docker installed through deb packages" >> $LOG_FILE 
+  fi
+
+  # Check sources.list exist
+  if [ ! -f "$FILE_SOURCES_LIST" ]; then
+    echo "$FILE_SOURCES_LIST does not exist" 2>&1 | tee -a $LOG_FILE 
+    exit 1
+  fi
+
+  # Check security.sources.list does not exist
+  if [ -f "$FILE_SECURITY_SOURCES_LIST" ]; then
+    echo "$FILE_SECURITY_SOURCES_LIST already exists" 2>&1 | tee -a $LOG_FILE 
+    exit 1
+  fi
+
+  # Create security.sources.list
+  echo -e "\e[32mCreating $FILE_SECURITY_SOURCES_LIST\e[0m" >> $LOG_FILE 
+  grep security "$FILE_SOURCES_LIST" | tee "$FILE_SECURITY_SOURCES_LIST" &>/dev/null || { echo "error creating file $FILE_SECURITY_SOURCES_LIST" 2>&1 | tee -a $LOG_FILE ; exit 1; }
+
+  # Prevent docker packages from been updated if docker installed via repositories
+  if [ "$DOCKER_INSTALLATION_REPOSITORIES" = true ] ; then
+    echo -e "\e[32mEditing docker repositories\e[0m" >> $LOG_FILE 
+    for DOCKER_PACKAGE in "${DOCKER_PACKAGES[@]}"; do
+      echo "${DOCKER_PACKAGE} hold" | sudo dpkg --set-selections
+    done
+  fi
+}
+
+
+
+##################
+#2.UPDATE-UPGRADE#
+##################
+
+function update () {
+  echo -e "\e[32mUpdating\e[0m" >> $LOG_FILE 
+  # 1>/dev/null 2>>$LOG_FILE: will redirect stdout to null and stderr to log file
+  apt-get update 1>/dev/null 2>>$LOG_FILE || { echo "error on apt-get update" 2>&1 | tee -a $LOG_FILE ; exit 1; }
```

Comment:
Why not log both stdin and stderr?

Reply:
AFAIK the important info would be just the error logs. When testing this script there were tons of logs saying a lot of stuff that I did not consider much important